src/main/kotlin/org/rust/ide/console/RsConsoleCodeFragmentContext.kt

```diff
@@ -57,15 +102,17 @@ class RsConsoleCodeFragmentContext {
 }
 
 class RsConsoleOneCommandContext(codeFragment: RsReplCodeFragment) {
-    val topLevelElements: MutableMap<String, String> = mutableMapOf()
-    val statementsText: String
+    val command: String
+    val itemsNames: List<String>
+    val containsUseDirective: Boolean
 
     init {
-        for (namedElement in codeFragment.namedElements) {
-            val name = namedElement.name ?: continue
-            topLevelElements[name] = namedElement.text
-        }
+        val elements = codeFragment.expandedStmtsAndTailExpr.first
+        command = elements.joinToString("\n") { it.text }
+
+        itemsNames = elements.filterIsInstance<RsNamedElement>()
```

Comment:
I think we don't need uniqueness here, code in `RsConsoleCodeFragmentContext` would work correctly if there are duplicates. Also if there are items with same name, then command will not be executed successfully, so `RsConsoleOneCommandContext` will not be used

Reply:
It just looks a bit confusing that there are two cases when all named elements are needed, but they are collected in different ways. Probably there could be something like:
```Kotlin
val namedElementsUnique: Map<String, RsNamedElement>
    get() = expandedStmtsAndTailExpr.first  // to take expanded named elements into account when you implement processing macros in `expandedStmtsAndTailExpr`
        .filterIsInstance<RsNamedElement>()
        .filter { it.name != null }
        .associateBy { it.name!! }
```
So this property can be used when named elements or just their names are needed. But this can be done later.

Reply:
Ok, I agree that it is confusing, so updated code as you proposed